backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/allocation/impl/AbstractProducerDestination.java

```diff
@@ -257,6 +260,12 @@ private I_M_HU createNewHU(final IAllocationRequest request)
 
 		DYNATTR_IsEmptyHU.setValue(hu, true);
 
+		if (request.getReference() != null && request.getReference().isOfType(I_M_HU.class))
+		{
+			huQrCodesService.propagateQrForSplitHUs(request.getReference().getModel(I_M_HU.class),
```

Comment:
pls avoid calling getModel(I_M_HU.class).

always use the DAO/repo,
i.e. handlingUnitsBL/handlingUnitsDAO.getById.

In this case i would suggest to introduce 
de.metas.handlingunits.qrcodes.service.HUQRCodesService.propagateQrForSplitHUs(@NonNull final HuId sourceHUId, @NonNull final ImmutableList<I_M_HU> splitHUs)